integration_tests/test_dbapi.py

```diff
@@ -87,22 +90,45 @@ def test_select_query_result_iteration_statement_params(presto_connection):
         params=(3,)  # expecting all the rows with id >= 3
     )
 
-def test_select_query_result_string_statement_params(presto_connection):
+def test_select_query_result_multi_type_statement_params(presto_connection):
     """
     Tests string parameters in prepared statements
     """
     cur = presto_connection.cursor()
+    expected_uuid = uuid.uuid1()
     # test a simple string parameter and
     # a parameter with a quote in it
     cur.execute(
-        "select ?",
-        params=("six'",),
+        "SELECT ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?",
```

Comment:
This is efficient (one query, many assertions), but this does not "scale" when you add more test cases (the list of question marks will grow and grow), or more assertions -- for example we will want to verify not only how `SELECT ?` looks like back on our side, but also what Presto engine things of the type of `?` (eg `SELECT typeof(?)`)

Thus, I would rather go with separate query + assertions for every value tested.
This could be similar to what we do with JDBC tests.
(I am linking an example to a read path test, but i am sure you get what i mean)
https://github.com/prestosql/presto/blob/cbcba450481feb3d7b92fb862dd6692b6a921f7e/presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java#L188-L202

Reply:
broke out into multiple tests, thanks for the tip!